src/msw/rt/notifmsgrt.cpp

```diff
@@ -174,8 +174,8 @@ class wxToastNotifMsgImpl : public wxNotificationMessageImpl
         if ( SUCCEEDED(hr) )
         {
             wxCOMPtr<IToastNotificationFactory> factory;
-            hr = rt::GetActivationFactory(RuntimeClass_Windows_UI_Notifications_ToastNotification,
-                IID_IToastNotificationFactory, reinterpret_cast<void**>(&factory));
+            hr = (rt::GetActivationFactory(RuntimeClass_Windows_UI_Notifications_ToastNotification,
```

Comment:
Sorry, I've somehow missed this change, but I don't understand why do we need this and why is it correct. [According to MSDN](https://docs.microsoft.com/en-us/cpp/cppcx/wrl/getactivationfactory-function), `GetActivationFactory()` already returns an `HRESULT`, and here we convert it to `ERROR_INVALID_FUNCTION` which is the same as `S_FALSE`, i.e. would pass `SUCCEEDED()` check below.

Could you please explain what's going on here? Does the actual function differ from its documented behaviour? If so, we may have a problem when/if it is changed to conform to it...

Reply:
`rt::GetActivationFactory` is our own function, and it returns a `boolean`. This is assigned to a `HRESULT` (`long`). That is why I added the explicit types.
Nonetheless, using `S_OK`  and `S_FALSE` are better to use here.

Reply:
Sorry, I totally confused it with `RoGetActivationFactory()`.